internal/controller/git/work_tree.go

```diff
@@ -460,6 +460,11 @@ type TagMetadata struct {
 	// Subject is the subject (first line) of the commit message associated
 	// with the tag.
 	Subject string
+	// Tagger is the person who created the tag, in the format "Name <email>".
+	// This field is only populated for annotated tags.
+	Tagger string
+	// Annotation is the annotation of the tag, if it is an annotated tag.
+	Annotation string
```

Comment:
Not a new problem, but I notice that this struct is missing a field for the `commitDate` of the commit pointed to by the tag.

Would now be a good time to rectify that?

Reply:
I am not sure what is best. The way Git deals with it, is that it has the concept of a "creator date" which will either point to the date of the commit or the date of the annotated tag. I.e., it gives you a single field that is the most detailed.

If we were to move away from this, I would think we would deprecate the `CreatorDate` (including in the `DiscoveredCommit` API object) in favor of a `CommitterDate` and `TaggerDate` (following Git idioms) or `CommitDate` and `TagDate` (following slightly more natural language).

Reply:
Ah... I'd not dug deep enough into this to realize that it wasn't simply something that was available and we forgot to read it.

I would say we can reconsider later if/how this can be improved.

I don't want that to hold this up.

LGTM